### PR TITLE
Moved Di definitions to new class based format

### PR DESCRIPTION
**Pull request by jakeasmith**

### CODE REVIEW

**Comment by harikt** [comment]
Thank you @jakeasmith

**Comment by harikt** [comment]
Hey dude, the json seems not valid. Could you check the composer validate and send a PR fixing the same ?

If not I will do .

Thank you.

File: composer.json [hunk 1]
```diff
@@ -26,12 +26,16 @@
     },
     "autoload": {
         "psr-4": {
-            "Aura\\Session\\": "src/"
+            "Aura\\Session\\": "src/",
+            "Aura\\Session\\_Config": "config/",
```
Comment (harikt) [inline composer.json:30]: the problem seems here.

File: config/Common.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+         * Aura\Session\Randval
+         */
+        $di->params['Aura\Session\Randval']['phpfunc'] = $di->lazyNew('Aura\Session\Phpfunc');
+
+        /**
+         * Aura\Session\Segment
+         */
+        $di->params['Aura\Session\Segment'] = [
+            'session' => $di->lazyGet('session'),
+        ];
```
Comment (harikt) [inline config/Common.php:40]: make it 5.3 compatible array . As di is 5.3 compatible.

File: config/Common.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+        /**
+         * Aura\Session\CsrfTokenFactory
+         */
+        $di->params['Aura\Session\CsrfTokenFactory']['randval'] = $di->lazyNew('Aura\Session\Randval');
+
+        /**
+         * Aura\Session\Session
+         */
+        $di->params['Aura\Session\Session'] = [
```
Comment (harikt) [inline config/Common.php:24]: same thoughts here.

**Comment by harikt** [comment]
Don't worry I am working on fixing the same.

**Comment by jakeasmith** [comment]
Whoops. Sorry about that. Thanks for taking care of it.

**Comment by harikt** [comment]
no problem. I didn't even looked that close. 

You are welcome.

**Pull request merged**